coap-core/src/test/java/com/mbed/coap/transport/udp/DatagramSocketTransportTest.java

```diff
@@ -66,61 +65,48 @@ public void initializingWithStateException() throws IOException {
 
             trans.start(mock(CoapReceiver.class));
 
-            try {
-                trans.setReuseAddress(true);
-                fail();
-            } catch (Exception e) {
-                assertTrue(e instanceof IllegalStateException);
-            }
-            try {
-                trans.setSocketBufferSize(1234);
-                fail();
-            } catch (Exception e) {
-                assertTrue(e instanceof IllegalStateException);
-            }
         } finally {
             trans.stop();
         }
     }
 
     @Test
     public void initializeWithParameters() throws Exception {
-        DatagramSocketTransport trans = new DatagramSocketTransport(new InetSocketAddress(0), mock(Executor.class));
-        trans.setReuseAddress(false);
-        trans.setSocketBufferSize(12345);
+        DatagramSocketAdapter datagramSocket = new DatagramSocketAdapter(new InetSocketAddress(0), false, 12345);
+        DatagramSocketTransport trans = new DatagramSocketTransport(datagramSocket, mock(Executor.class));
         trans.start(mock(CoapReceiver.class));
 
-        assertTrue(trans.getSocket().isBound());
-        assertFalse(trans.getSocket().isClosed());
-        assertEquals(12345, trans.getSocket().getSendBufferSize());
-        assertFalse(trans.getSocket().getReuseAddress());
+        assertTrue(datagramSocket.isBound());
+        assertFalse(datagramSocket.isClosed());
+        assertEquals(12345, datagramSocket.getSendBufferSize());
+        assertFalse(datagramSocket.getReuseAddress());
 
         trans.stop();
-        assertTrue(trans.getSocket().isClosed());
+        assertTrue(datagramSocket.isClosed());
     }
 
     @Test
     public void reopenSamePort() throws IOException {
```

Comment:
I think this test does not make sense any more, just remove it.

Reply:
removed.